docs/examples/howto/add_new_algorithm.rst

```diff
@@ -0,0 +1,174 @@
+    `Examples/Framework/include/ActsExamples/Framework`.
+
+.. important:: The constructor should ideally follow certain rules. See section on configuration below.
+
+The `execute` method will be called once for each event. 
+Obviously other methods can be also added to this class.
+It is good to remember that the algorithmic code in ACTS is typically stateless
+and if the state needs to be passed between the calls it should be done explicitly.
+For instance if the event processing is best organized with a methods:
+
+.. code-block:: cpp
+
+    void prepareBuffers();
+    void fillBuffers( const SpacePoint& );
+    void extractInfo();
+
+that need to pass the data between each other these methods should rather look like this:
+
+.. code-block:: cpp
+
+    struct SeedingBuffers{ ... some buffers ... };
+    void prepareBuffers(SeedingBuffers& buffers);
+    void fillBuffers(SeedingBuffers& buffers, const SpacePoint& );
+    void extractInfo(const SeedingBuffers& buffers);
+
+
+..  tip:: It is a common practice to put the algorithm code in ``ActsExamples`` namespace.
+
+Input and Output
+------------------
+
+The algorithm would be typically part of some processing sequence and thus consume and produce some event data.
+In hypothetical example discussed here space-points cod be the input and track seeds would be an output.
+The data can be retrieved in the algorithm using `get` method.
+
+.. code-block:: cpp
+
+    template<typename T>
+    const T& get(const std::string& name);
+
+The data is fetched from so called "store" that is populated by preceding algorithm or by the reader.
+
+The data object (or objects) produced by an algorithm can be placed in the store using method `add`.
+
+.. code-block:: cpp
+
+    template<typename T>
+    void add(const std::string& name, T&& object);
+
+As the method signature suggests the ownership of the object is transferred to the store.
+That is, the destruction of this object at the end of event processing is taken care of.
+
+Configurability
+----------------
+
+It is customary that an algorithm requires configuration parameters.
+For the sake of an example in hypothetical seeding algorithm it could be parameter steering
+which detector layers should be used. 
+The configuration can be provided to an algorithm through an additional class/structure.
+It can be an inner class of the algorithm or it can be external to it.
+One would use definitely external structure if it is shared among several algorithm classes.
+
+For example that is how the configuration object could look like for `MySeedingAlgorithm`:
+
+.. code-block:: cpp
+
+    struct MySeedingConfig {
+        std::vector<int> layers; // layers to use by the seeder
+        float deltaZ;  // the maximum allowed deviation in r-z plane
+    };
+
+.. tip:: It is customary to put the config structures in ``Acts`` namespace.
+
+The algorithm constructor would take `MySeedingConfig` object during the construction in addition 
```

Comment:
The algorithm constructor would take a MySeedingConfig object during the construction in addition to an argument controlling verbosity of diagnostic messages.